beacon-chain/rpc/beaconv1/blocks.go

```diff
@@ -70,11 +70,15 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpb.BlockHeadersR
 			return nil, status.Errorf(codes.Internal, "Could not retrieve blocks: %v", err)
 		}
 	} else {
-		_, blks, err = bs.BeaconDB.BlocksBySlot(ctx, req.Slot)
+		slot := bs.ChainInfoFetcher.HeadSlot()
+		if req.Slot != nil {
+			slot = *req.Slot
+		}
+		_, blks, err = bs.BeaconDB.BlocksBySlot(ctx, slot)
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "Could not retrieve blocks for slot %d: %v", req.Slot, err)
 		}
-		_, blkRoots, err = bs.BeaconDB.BlockRootsBySlot(ctx, req.Slot)
+		_, blkRoots, err = bs.BeaconDB.BlockRootsBySlot(ctx, slot)
```

Comment:
This looks like a logic change. Can you explain?

Reply:
Since req.Slot is now a pointer, changing the api endpoint to handle it properly for now.

Reply:
This doesn't make any sense though, you now only fetch from the head slot which renders any data in the request pointless. Whatever a user requests, they will still get the same data.

Reply:
Wdym? I check if  the request they sent is nil and then pass in that info instead. Why would it always be head slot?

Reply:
Nvm ignore the comment, was looking at something else and got it mixed with this.